server/src/test/java/org/candlepin/resteasy/JsonProviderTest.java

```diff
@@ -142,4 +148,203 @@ private boolean isEnabled(JsonProvider provider, SerializationFeature feature) {
+            fail("Parsing serialized ActivationKeyDTO failed!");
+        }
+        assertEquals(1, akNode.size());
+        assertTrue(akNode.has("name"), "The 'name' field should have been included!");
+    }
+
+    @Test
+    public void testDynamicPropertyFilterIncludeMultipleProperties() {
+        DynamicFilterData filterData = new DynamicFilterData();
+        filterData.includeAttribute("name");
+        filterData.includeAttribute("releaseVer");
+        filterData.includeAttribute("addOns");
+        filterData.setWhitelistMode(true); // When only includes are set, we should be in whitelist mode
+        ResteasyProviderFactory.pushContext(DynamicFilterData.class, filterData);
+
+        ActivationKeyDTO keyDTO = new ActivationKeyDTO();
+        String serializedKey = "";
+        try {
+            serializedKey = ourMapper.writeValueAsString(keyDTO);
+        }
+        catch (JsonProcessingException e) {
+            fail("Serializing ActivationKeyDTO failed!");
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode akNode = null;
+        try {
+            akNode = mapper.readTree(serializedKey);
+        }
+        catch (JsonProcessingException e) {
+            fail("Parsing serialized ActivationKeyDTO failed!");
+        }
+        assertEquals(3, akNode.size());
+        assertTrue(akNode.has("name"), "The 'name' field should have been included!");
+        assertTrue(akNode.has("releaseVer"), "The 'releaseVer' field should have been included!");
+        assertTrue(akNode.has("addOns"), "The 'addOns' field should have been included!");
+    }
+
+    @Test
+    public void testDynamicPropertyFilterIncludeNestedProperty() {
+        DynamicFilterData filterData = new DynamicFilterData();
+        filterData.includeAttribute("owner.id");
+        filterData.setWhitelistMode(true); // When only includes are set, we should be in whitelist mode
+        ResteasyProviderFactory.pushContext(DynamicFilterData.class, filterData);
+
+        ActivationKeyDTO keyDTO = new ActivationKeyDTO();
+        NestedOwnerDTO ownerDTO = new NestedOwnerDTO()
+            .key("owner_key")
+            .id("owner_id");
+        keyDTO.setOwner(ownerDTO);
+
+        String serializedKey = "";
+        try {
+            serializedKey = ourMapper.writeValueAsString(keyDTO);
+        }
+        catch (JsonProcessingException e) {
+            fail("Serializing ActivationKeyDTO failed!");
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode akNode = null;
+        try {
+            akNode = mapper.readTree(serializedKey);
+        }
+        catch (JsonProcessingException e) {
+            fail("Parsing serialized ActivationKeyDTO failed!");
+        }
+        assertEquals(1, akNode.size());
+        assertTrue(akNode.has("owner"), "The 'owner' field should have been included!");
+        assertTrue(akNode.get("owner").has("id"), "The 'owner.id' field should have been included!");
```

Comment:
For these tests where we have nested nodes, we should also verify that the children nodes have the correct amount of elements.

Applies below as well

Reply:
Done